assets/js/modules/adsense/datastore/settings.js

```diff
@@ -0,0 +1,444 @@
+			const { params } = payload;
+			const { useSnippet } = params;
+
+			return {
+				...state,
+				// Update saved settings.
+				savedSettings: {
+					...( state.savedSettings || {} ),
+					useSnippet,
+				},
+				// Also update client settings to ensure they're in sync.
+				settings: {
+					...( state.settings || {} ),
+					useSnippet,
+				},
+			};
+		}
+
+		case CATCH_FETCH_SAVE_USE_SNIPPET: {
+			const { error } = payload;
+
+			return {
+				...state,
+				error,
+				isFetchingSaveUseSnippet: false,
+			};
+		}
+
+		case START_SUBMIT_CHANGES: {
+			return {
+				...state,
+				isDoingSubmitChanges: true,
+			};
+		}
+
+		case FINISH_SUBMIT_CHANGES: {
+			return {
+				...state,
+				isDoingSubmitChanges: false,
+			};
+		}
+
+		// This action is purely for testing, the value is typically handled
+		// as a side-effect from 'RECEIVE_SETTINGS' (see below).
+		case RECEIVE_ORIGINAL_ACCOUNT_STATUS: {
+			const { originalAccountStatus } = payload;
+			return {
+				...state,
+				originalAccountStatus,
+			};
+		}
+
+		// This action is mainly handled via createSettingsStore, but here we
+		// need it to have the side effect of storing the original account
+		// status.
+		case 'RECEIVE_SETTINGS': {
+			const { values } = payload;
+			const { accountStatus } = values;
+
+			// Only set original account status when it is really the first
+			// time that we load the settings on this pageload.
+			if ( 'undefined' === typeof state.originalAccountStatus ) {
+				return {
+					...state,
+					originalAccountStatus: accountStatus,
+				};
+			}
+
+			return { ...state };
+		}
+
+		default: {
+			return { ...state };
+		}
+	}
+};
+
+export const resolvers = {
+	*getOriginalAccountStatus() {
+		const registry = yield commonActions.getRegistry();
+
+		// Do not do anything if original account status already known.
+		const existingOriginalAccountStatus = registry.select( STORE_NAME ).getOriginalAccountStatus();
+		if ( 'undefined' !== typeof existingOriginalAccountStatus ) {
+			return;
+		}
+
+		// Ensure settings are being fetched if not yet in progress.
+		registry.select( STORE_NAME ).getSettings();
+	},
+};
+
+export const selectors = {
+	/**
+	 * Checks if changes can be submitted.
+	 *
+	 * @since n.e.x.t
+	 *
+	 * @return {boolean} True if changes can be submitted, false otherwise.
+	 */
+	canSubmitChanges: createRegistrySelector( ( select ) => () => {
+		const {
+			getAccountID,
+			getClientID,
+			getAccountStatus,
+			haveSettingsChanged,
+			isDoingSubmitChanges,
+		} = select( STORE_NAME );
+
+		if ( isDoingSubmitChanges() ) {
+			return false;
+		}
+		if ( ! haveSettingsChanged() ) {
+			return false;
+		}
+		// Require an account status to be present.
+		if ( ! getAccountStatus() ) {
+			return false;
+		}
+		// Require account ID to be either empty (if impossible to determine)
+		// or valid.
+		const accountID = getAccountID();
+		if ( '' !== accountID && ! isValidAccountID( accountID ) ) {
```

Comment:
This is a very specific check here, an empty string is not a valid account ID, so that shouldn't be baked into the function. However, we need to allow for an empty string to be saved, since there are cases where no account ID (same for client ID) can be determined.